psydac/api/ast/linalg.py

```diff
@@ -156,9 +156,17 @@ def _initialize(self, ndim, block_shape, **kwargs):
         interface_axis  = kwargs.pop('interface_axis', None)
         d_start         = kwargs.pop('d_start', None)
         c_start         = kwargs.pop('c_start', None)
-        mats            = [variables('mat{}'.format(''.join(str(i) for i in key)),'real', cls=IndexedVariable, rank=2*ndim) for key in keys]
-        xs              = [variables('x{}'.format(i),'real', cls=IndexedVariable, rank=ndim) for i in range(block_shape[1])]
-        outs            = [variables('out{}'.format(i),'real', cls=IndexedVariable, rank=ndim) for i in range(block_shape[0])]
+        dtype           = kwargs.pop('dtype', float)
+
+        # Adapt the type of data treated in our dot function
+        if dtype==complex:
+            dtype_string='complex'
+        else:
+            dtype_string='real'
```

Comment:
Pyccel support for `'real'` may be dropped in the future. Please use `'float'` instead.

Reply:
When I try with float it return an error "ValueError : Unrecognized datatype float"

Reply:
I see. This string must be used by the (partial and old) version of Pyccel which is inside `psydac.pyccel`. Please disregard my comment.